src/tree.h

```diff
@@ -42,6 +42,8 @@ extern int git_tree_entry_icmp(const git_tree_entry *e1, const git_tree_entry *e
 void git_tree__free(void *tree);
 int git_tree__parse(void *tree, git_odb_object *obj);
 
+int git_treebuilder_write_with_buffer(git_oid *oid, git_treebuilder *bld, git_buf *tree);
```

Comment:
The other direction to go to would be to make it part of the public API. This declaration would then be moved to `include/git2/tree.h` and would need some documentation. I'd be in favour of making it public as it can be useful for external users of the tree builder as well.